syntax = "proto3";

import "scanner/types.proto";
package scanner.proto;

message BlurArgs {
  int32 kernel_size = 1;
  float sigma = 2;
}

message MontageArgs {
  int64 num_frames = 1;
  int32 target_width = 4;
  int32 frames_per_row = 6;
}

message CaffeInputArgs {
  NetDescriptor net_descriptor = 1;
  int32 batch_size = 2;
}

message CaffeArgs {
  NetDescriptor net_descriptor = 1;
  int32 batch_size = 2;
}

message FacenetArgs {
  CaffeArgs caffe_args = 1;
  float scale = 2;
  float threshold = 3;
}

message CPM2Args {
  CaffeArgs caffe_args = 1;
  float scale = 2;
}

message Camera {
  repeated float p = 1 [packed=true];
}

message GipumaArgs {
  float min_disparity = 2;
  float max_disparity = 3;
  float min_depth = 4;
  float max_depth = 5;
  float iterations = 6;
  int32 kernel_width = 7;
  int32 kernel_height = 8;
}

enum ExtractorType {
  SIFT = 0;
  SURF = 1;
}

message FeatureExtractorArgs {
  ExtractorType feature_type = 1;
}

message Keypoint {
  float x = 1;
  float y = 2;
}

message ResizeArgs {
  int32 width = 1;
  int32 height = 2;
  bool min = 3;
  bool preserve_aspect = 4;
}

message ImageDecoderArgs {
  enum ImageType {
    JPEG = 0;
    ANY = 1;
  }

  ImageType image_type = 1;
}
